client/servers/manager_test.go

```diff
@@ -186,41 +155,28 @@ func TestServers_NotifyFailedServer(t *testing.T) {
 	// Fail the other server
 	m.NotifyFailedServer(second)
 	next := m.FindServer()
-	if !next.Equal(first) {
-		t.Fatalf("Expected first server (still)")
-	}
+	must.True(t, next.Equal(first))
```

Comment:
I think you can use `must.Equal` here, as that calls the `Equal` method (as opposed to `must.Eq` which uses `cmp.Equal`).